lib/SIL/IR/SILBuilder.cpp

```diff
@@ -241,6 +257,9 @@ void SILBuilder::emitBlock(SILBasicBlock *BB, SILLocation BranchLoc) {
 /// instruction) then split the block at that instruction and return the
 /// continuation block.
 SILBasicBlock *SILBuilder::splitBlockForFallthrough() {
+#ifndef NDEBUG
+  PrevDebugScope = nullptr;
```

Comment:
I'm not sure this is the answer you're looking for, but since we only verify the nesting within a basic block, splitting should always make violations less likely than before.